src/puppetlabs/puppetdb/client.clj

```diff
@@ -126,3 +179,5 @@
                    ssl-opts)]
      (when-not (= HttpURLConnection/HTTP_OK (:status result))
        (log/error result)))))
+
+(set! *warn-on-reflection* warn-on-reflection-orig)
```

Comment:
Was this something you added while debugging or is it helpful to keep?

Reply:
So (particularly given the focus on performance), Austin and I have been trying out the idea of just keeping reflection eliminated in namespaces where the required changes aren't "too noisy".

There's no intention to do this eagerly, just whenever we like (and are presumably already making changes), and then we'll just leave the warnings turned on for those namespaces so we'll notice regressions.